File: app/src/main/java/com/amarnehsoft/holyquran/fragments/verses/VersesFragmentViewModel.java
package com.amarnehsoft.holyquran.fragments.verses;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.amarnehsoft.holyquran.Repo;
import com.amarnehsoft.holyquran.model.Surah;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.Single;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.observers.DisposableSingleObserver;
import io.reactivex.schedulers.Schedulers;

public class VersesFragmentViewModel extends ViewModel{

    private MutableLiveData<List<Integer>> resultLive = new MutableLiveData<>();
    private MutableLiveData<String> error = new MutableLiveData<>();
    public MutableLiveData<Boolean> showProgress = new MutableLiveData<>();

    private int numberOfVerses = 0;
    private int numberOfStartAya;

    @Inject
    public VersesFragmentViewModel(){

    }

    public void init(){
        showProgress.postValue(true);
        this.numberOfVerses = numberOfVerses;
        Single.just(numberOfVerses)
                .map(integer -> {
                    List<Integer> list = new ArrayList<>();
                    for (int i=1; i <= integer; i++){
                        list.add(i);
                    }
                    return list;
                }).subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new DisposableSingleObserver<List<Integer>>() {
                    @Override
                    public void onSuccess(List<Integer> integers) {
                        resultLive.postValue(integers);
                        showProgress.postValue(false);
                    }

                    @Override
                    public void onError(Throwable e) {
                        showProgress.postValue(false);
                        error.postValue(e.getMessage());
                    }
                });
    }

    public MutableLiveData<List<Integer>> getResultLive() {
        return resultLive;
    }

    public MutableLiveData<String> getError() {
        return error;
    }

    public int getNumberOfVerses() {
        return numberOfVerses;
    }

    public void setNumberOfVerses(int numberOfVerses) {
        this.numberOfVerses = numberOfVerses;
    }

    public void setNumberOfStartAya(int numberOfStartAya) {
        this.numberOfStartAya = numberOfStartAya;
    }

    public int getNumberOfStartAya() {
        return numberOfStartAya;
    }
}

File: app/src/main/java/com/amarnehsoft/holyquran/fragments/verses/VersesFragmentViewModel.kt
package com.amarnehsoft.holyquran.fragments.verses

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import java.util.ArrayList

import javax.inject.Inject

import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.observers.DisposableSingleObserver
import io.reactivex.schedulers.Schedulers

class VersesFragmentViewModel @Inject
constructor() : ViewModel() {

    val resultLive = MutableLiveData<List<Int>>()
    val error = MutableLiveData<String>()
    var showProgress = MutableLiveData<Boolean>()

    var numberOfVerses = 0
    var numberOfStartAya: Int = 0

    fun init() {
        showProgress.postValue(true)
        this.numberOfVerses = numberOfVerses
        Single.just(numberOfVerses)
            .map<List<Int>> { integer ->
                val list = ArrayList<Int>()
                for (i in 1..integer) {
                    list.add(i)
                }
                list
            }.subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(object : DisposableSingleObserver<List<Int>>() {
                override fun onSuccess(integers: List<Int>) {
                    resultLive.postValue(integers)
                    showProgress.postValue(false)
                }

                override fun onError(e: Throwable) {
                    showProgress.postValue(false)
                    error.postValue(e.message)
                }
            })
    }
}
